server/src/lib.rs

```diff
@@ -0,0 +1,154 @@
+pub enum Backend {
+    Dynamo(aws_sdk_dynamodb::Client),
+    Local(Arc<Mutex<Local>>),
+}
+
+impl Backend {
+    #[allow(unused)]
+    async fn local() -> Self {
+        Backend::Local(Arc::new(Mutex::new(Local::default())))
+    }
+
+    /// Instantiate a DynamoDB backend.
+    ///  
+    /// If `USE_DYNAMODB` is set to "local", the `AWS_ENDPOINT_URL` will be taken
+    /// from the environment with the "http://localhost:8000" fallback , the `AWS_DEFAULT_REGION`
+    /// will be pulled from the environment as well and will default to "us-east-1",
+    /// as for the credentials - the [test credentials](https://docs.rs/aws-config/latest/aws_config/struct.ConfigLoader.html#method.test_credentials)
+    /// will be used to sign requests.
+    ///
+    /// This spares setting those environment variables (including `AWS_ACCESS_KEY_ID`
+    /// and `AWS_SECRET_ACCESS_KEY`) via the command line or configuration files,
+    /// and allows to run the application against a local dynamodb instance with just:
+    /// ```sh
+    /// USE_DYNAMODB=local cargo run
+    /// ```
+    /// While the entire test suite can be run with:
+    /// ```sh
+    /// USE_DYNAMODB=local cargo t -- --include-ignored
+    /// ```
+    ///
+    /// This also allows us to use the local instance of DynamoDB which is running
+    /// in a container on the same network, in which case the database will be accessible
+    /// under `http://<dynamodb_container_name>:<port>`. This facilitates the setup of
+    /// local API Gateway with SAM, since the `sam local start-api` command will launch our
+    /// back-end app in a docker container.
+    ///
+    /// If more customization is needed (say, you want to set some specific credentials
+    /// rather than rely on those test creds generated by the `aws_config` crate),
+    /// set `USE_DYNAMODB` to e.g. "custom", and set the environment variables to whatever
+    /// values you need or let them be picked up from your `~/.aws` files
+    /// (see [`aws_config::load_from_env`](https://docs.rs/aws-config/latest/aws_config/fn.load_from_env.html))
+    pub async fn dynamo() -> Self {
+        let config = if std::env::var("USE_DYNAMODB")
+            .ok()
+            .is_some_and(|v| v == "local")
+        {
+            aws_config::from_env()
+                .endpoint_url(
+                    std::env::var("AWS_ENDPOINT_URL")
+                        .ok()
+                        .unwrap_or("http://localhost:8000".into()),
+                )
+                .region(aws_config::Region::new(
+                    std::env::var("AWS_DEFAULT_REGION")
+                        .ok()
+                        .unwrap_or("us-east-1".into()),
+                ))
+                .test_credentials()
+                .load()
+                .await
+        } else {
+            aws_config::load_from_env().await
+        };
+        Backend::Dynamo(aws_sdk_dynamodb::Client::new(&config))
+    }
+}
+
+pub async fn build_app() -> Router {
```

Comment:
The name `build_app` is a little weird to me. How about just `new`? So `wewerewondering::new()` returns the `Router`?

Reply:
Made one small adjustment here: https://github.com/jonhoo/wewerewondering/pull/218/commits/2f17ca4a0058c5df6dc26f17639d0cd147ff217f